logger_test.go

```diff
@@ -203,3 +162,27 @@ func TestLogData(t *testing.T) {
 
 	close(done)
 }
+
+func TestAppendData(t *testing.T) {
+	clearBackends()
+	defer resetOutput(setBuffer())
+
+	comm := RegisterBackend()
```

Comment:
Since we are testing the logger here and not the backend, and since these are unit-tests, I suggest 'using' the fact that we can access internal logger structure here and simply iterating over the data slice. 
Alternatively, we can access e.Data and iterate on that. But I still feel that would be too much for an AppendData unit-test.
I admit: I didn't look at the tests before; the missing ellipsis made me do it ;)

Reply:
That wouldn't exercise the usage of the logging functionality, and doesn't test the actual use case, just the structure of the logger. If we wanted to change how the data was stored later, we'd need to change the tests.

Reply:
I don't think the updated test would catch the missing ellipsis either (because the string would still be contained, it just wouldn't be appended to the data slice properly). But I guess we can make refactoring the tests a separate PR as it's too much work.